### PR TITLE
[4.0] Plugins using more namespaced classes Part 1

### PR DESCRIPTION
**Pull request by anuragteapot**
### Summary of Changes

Switch to the namespaced classes

### Testing Instructions
Simple code review

### CODE REVIEW

**Comment by anuragteapot** [comment]
@C-Lodder Please review changes

**Comment by joomdonation** [comment]
Could you please change JPlugin to the correct namespace class Joomla\CMS\Plugin\CMSPlugin?

**Comment by anuragteapot** [comment]
@joomdonation sure Thanks for review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19309">issues.joomla.org/tracker/joomla-cms/19309</a>.</sub>

File: plugins/authentication/cookie/cookie.php [hunk 1]
```diff
@@ -147,9 +155,9 @@ public function onUserAuthenticate($credentials, $options, &$response)
 			catch (RuntimeException $e)
```
Comment (C-Lodder) [inline plugins/authentication/cookie/cookie.php:155]: @wilsonge should `RuntimeException` have a `\` before?

File: plugins/captcha/recaptcha/recaptcha.php [hunk 2]
```diff
@@ -45,15 +48,15 @@ public function onInit($id = 'dynamic_recaptcha_1')
 
 		if ($pubkey === '')
 		{
-			throw new Exception(JText::_('PLG_RECAPTCHA_ERROR_NO_PUBLIC_KEY'));
+			throw new Exception(Text::_('PLG_RECAPTCHA_ERROR_NO_PUBLIC_KEY'));
```
Comment (C-Lodder) [inline plugins/captcha/recaptcha/recaptcha.php:52]: Backslash before `Exception` @wilsonge ?

File: plugins/authentication/cookie/cookie.php [hunk 3]
```diff
@@ -13,6 +13,7 @@
 use Joomla\CMS\User\User;
 use Joomla\CMS\Language\Text;
 use Joomla\CMS\User\UserHelper;
+use Joomla\CMS\Plugin\CMSPlugin;
 use Joomla\CMS\Filter\InputFilter;
 use Joomla\CMS\Helper\AuthenticationHelper;
```
Comment (joomdonation) [inline plugins/authentication/cookie/cookie.php]: use Joomla\CMS\Helper\AuthenticationHelper; is not needed as it is not used

File: plugins/authentication/gmail/gmail.php [hunk 4]
```diff
@@ -14,6 +14,7 @@
 use Joomla\CMS\Language\Text;
 use Joomla\Registry\Registry;
 use Joomla\CMS\Http\HttpFactory;
+use Joomla\CMS\Plugin\CMSPlugin;
 use Joomla\CMS\Helper\AuthenticationHelper;
```
Comment (joomdonation) [inline plugins/authentication/gmail/gmail.php]: use Joomla\CMS\Helper\AuthenticationHelper; is not needed as it is not used

File: plugins/authentication/ldap/ldap.php [hunk 5]
```diff
@@ -11,6 +11,7 @@
 
 use Joomla\Ldap\LdapClient;
 use Joomla\CMS\Language\Text;
+use Joomla\CMS\Plugin\CMSPlugin;
 use Joomla\CMS\Helper\AuthenticationHelper;
```
Comment (joomdonation) [inline plugins/authentication/ldap/ldap.php]: use Joomla\CMS\Helper\AuthenticationHelper; is not needed as it is not used

File: plugins/editors/tinymce/field/skins.php [hunk 6]
```diff
@@ -10,10 +10,12 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Form\FormHelper;
+use Joomla\CMS\Form\Field\ListField;
 
 jimport('joomla.form.helper');
 
-JFormHelper::loadFieldClass('list');
+FormHelper::loadFieldClass('list');
```
Comment (joomdonation) [inline plugins/editors/tinymce/field/skins.php]: I am unsure 100% but I think this command could be removed. The Joomla\CMS\Form\Field\ListField class should be autoload-already. Could you please help testing to see remove it works? If yes, please remove it.

File: plugins/editors/codemirror/fonts.php [hunk 7]
```diff
@@ -11,8 +11,10 @@
 defined('_JEXEC') or die;
 
 use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Form\FormHelper;
+use Joomla\CMS\Form\Field\ListField;
 
-JFormHelper::loadFieldClass('list');
+FormHelper::loadFieldClass('list');
```
Comment (joomdonation) [inline plugins/editors/codemirror/fonts.php]: I am unsure 100% but I think this command could be removed. The Joomla\CMS\Form\Field\ListField class should be autoload-already. Could you please help testing to see remove it works? If yes, please remove it.

File: plugins/editors/tinymce/field/uploaddirs.php [hunk 8]
```diff
@@ -10,11 +10,13 @@
...
 
 use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Form\FormHelper;
 use Joomla\CMS\Component\ComponentHelper;
+use Joomla\CMS\Form\Field\FolderlistField;
 
 jimport('joomla.form.helper');
 
-JFormHelper::loadFieldClass('folderlist');
+FormHelper::loadFieldClass('folderlist');
```
Comment (joomdonation) [inline plugins/editors/tinymce/field/uploaddirs.php]: I am unsure 100% but I think this command could be removed. The Joomla\CMS\Form\Field\FolderlistField class should be autoload already. Could you please help testing to see remove it works? If yes, please remove it.

File: plugins/authentication/cookie/cookie.php (hunk 1 above)
Comment (joomdonation) [reply plugins/authentication/cookie/cookie.php:155]: At this time, I think it is not needed as the plugin is not namespaced yet.

File: plugins/authentication/cookie/cookie.php (hunk 3 above)
Comment (anuragteapot) [reply plugins/authentication/cookie/cookie.php]: But, without <code>Joomla\CMS\Helper\AuthenticationHelper</code> I am getting error.

File: plugins/editors/codemirror/fonts.php (hunk 7 above)
Comment (anuragteapot) [reply plugins/editors/codemirror/fonts.php]: Ok, let me test.

File: plugins/authentication/cookie/cookie.php (hunk 3 above)
Comment (joomdonation) [reply plugins/authentication/cookie/cookie.php]: Strange as that AuthenticationHelper class is not used anywhere inside the plugin. I just tried to remove it here and don't see error. Could you post the error message you received?

Comment (anuragteapot) [reply plugins/authentication/cookie/cookie.php]: @joomdonation  Yes, Joomla\CMS\Helper\AuthenticationHelper;  is not needed here. I  review code. 
Thanks

File: plugins/editors/codemirror/fonts.php [hunk 9]
```diff
@@ -10,7 +10,11 @@
 // No direct access
 defined('_JEXEC') or die;
 
-JFormHelper::loadFieldClass('list');
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Form\FormHelper;
+use Joomla\CMS\Form\Field\ListField;
+
+FormHelper::loadFieldClass('list');
```
Comment (joomdonation) [inline plugins/editors/codemirror/fonts.php]: Please remove this line, it is not needed

File: plugins/editors/codemirror/fonts.php [hunk 10]
```diff
@@ -10,7 +10,11 @@
 // No direct access
 defined('_JEXEC') or die;
 
-JFormHelper::loadFieldClass('list');
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Form\FormHelper;
```
Comment (joomdonation) [inline plugins/editors/codemirror/fonts.php]: Please remove this line, it is not needed

File: plugins/editors/tinymce/field/skins.php [hunk 11]
```diff
@@ -9,9 +9,13 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Form\FormHelper;
```
Comment (joomdonation) [inline plugins/editors/tinymce/field/skins.php]: Please remove this line, it is not needed

File: plugins/editors/tinymce/field/skins.php [hunk 12]
```diff
@@ -9,9 +9,13 @@
...
+use Joomla\CMS\Form\Field\ListField;
+
 jimport('joomla.form.helper');
 
-JFormHelper::loadFieldClass('list');
+FormHelper::loadFieldClass('list');
```
Comment (joomdonation) [inline plugins/editors/tinymce/field/skins.php]: Please remove this line, it is not needed

File: plugins/editors/tinymce/field/uploaddirs.php [hunk 13]
```diff
@@ -9,9 +9,14 @@
 
 defined('_JEXEC') or die;
 
+use Joomla\CMS\HTML\HTMLHelper;
+use Joomla\CMS\Form\FormHelper;
```
Comment (joomdonation) [inline plugins/editors/tinymce/field/uploaddirs.php]: Same here, remove this line

File: plugins/editors/tinymce/field/uploaddirs.php [hunk 14]
```diff
@@ -9,9 +9,14 @@
...
+use Joomla\CMS\Component\ComponentHelper;
+use Joomla\CMS\Form\Field\FolderlistField;
+
 jimport('joomla.form.helper');
 
-JFormHelper::loadFieldClass('folderlist');
+FormHelper::loadFieldClass('folderlist');
```
Comment (joomdonation) [inline plugins/editors/tinymce/field/uploaddirs.php]: Please remove this line

**Comment by joomdonation** [comment]
Overall, it looks good. There are just some unnecessary code left in several places which can be removed as I commented

use Joomla\CMS\Form\FormHelper;
FormHelper::loadFieldClass

Once you removed it, I will mark my test result as success

**Comment by anuragteapot** [comment]
@joomdonation done!

**Comment by joomdonation** [comment]
I have tested this item :white_check_mark: successfully on dce5bc3d7a2ff5e88ad61ff137c5d067e71f8535<br /><br />Thanks for working on this.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19309">issues.joomla.org/tracker/joomla-cms/19309</a>.</sub>

**Comment by anuragteapot** [comment]
@C-Lodder review this also and part 3
Thanks

**Comment by C-Lodder** [comment]
I have tested this item :white_check_mark: successfully on dce5bc3d7a2ff5e88ad61ff137c5d067e71f8535<br /><br />code review<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/19309">issues.joomla.org/tracker/joomla-cms/19309</a>.</sub>

**Comment by anuragteapot** [comment]
@wilsonge please, take a look at this PR it contains big changes, any major change to base branch cause a major conflict to this branch.
Thanks :)

File: plugins/authentication/cookie/cookie.php (hunk 1 above)
Comment (wilsonge) [reply plugins/authentication/cookie/cookie.php:155]: What @joomdonation said :)

**Comment by wilsonge** [comment]
Sorry for being slow with this! The post christmas rush has been crazy in the office. Slowly catching up on everything

**Comment by anuragteapot** [comment]
@wilsonge Its, ok. :)
Thanks, currently I am preparing for GSOC 2018 with Joomla. That's why I am trying to fix bugs and contribute to Joomla as many I can :)

**Comment by anuragteapot** [comment]
#19324 #19323

**Comment by C-Lodder** [comment]
We've got lots of Javascript work if you're interested :P

**Comment by anuragteapot** [comment]
@C-Lodder yeah I am interested to work in Javascript :)

**Comment by anuragteapot** [comment]
Is any group where Joomla developers discuss.

**Comment by C-Lodder** [comment]
The Javascript Channel on Glip

**Comment by anuragteapot** [comment]
How can I join?

**Comment by C-Lodder** [comment]
@franz-wohlkoenig do you know who can set him up with an invite?

**Comment by ghost** [comment]
@C-Lodder Luca Marzo. Give Luca the E-Mail-Address of @Anu1601CS.

**Comment by anuragteapot** [comment]
Thanks for adding me to group.

**Comment by C-Lodder** [comment]
You should be able to see the Javascript group. If not let me know and I'll send you an invite

**Comment by anuragteapot** [comment]
I am not  able to find JavaScript group.

**Comment by C-Lodder** [comment]
I'll get Dimitris to add you

**Pull request merged**